### PR TITLE
DockedMegaMenu: Clean up toggle and old code

### LINKED ISSUE #72446: Epic: [Navigation] Megamenu with open/closed states
As we continue to improve Grafana's usability, we've received feedback indicating that navigation between certain features has become more difficult due to the introduction of the hamburger menu and the section navigation. There is also an increased effort required to access certain features that are now placed lower in the hierarchy. To address this, we propose refining the navigation system to enhance user experience and make features easier to access.

### User Stories

1. **Persistent Navigation:** I want to have the navigation visible as I move between pages, for tasks like gathering incident data, setting up Grafana for the first time, or exploring Grafana as a new user in an existing organization. 
2. **Collapsible Navigation:** I want to be able to collapse the navigation when I'm focused on a single task, like editing a dashboard.

### Acceptance Criteria
**Functional:**

1. Open/Closed State: The navigation sidebar can be collapsed.
2. 3 Levels: Top, second and third level navigation items are accessible from the megamenu.

**Responsiveness:**

1. Viewport Adaptability: Navigation is collapsed by default on viewports narrower than 1440px (or smaller?).
2. Mobile Devices: On mobile devices (viewports narrower than 768px), when opened, navigation is full screen.

**Accessibility:**

1. Touch/Click Targets: Touch/click targets meet WCAG AA requirements ([minimum 24x24 px](https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html)).
2. State Feedback: Navigation items come with hover and active states.

### UI Implications

1. Megamenu Sizing: As the megamenu will have more content than before, it needs to be compact. Current width is 350px, and proof-of-concepts suggest reducing this to 250px.
2. WCAG Compliance: Smaller nav item height while remaining WCAG AA compliant, with a minimum height of 24px.
3. Icon Selection: For desktop, select an appropriate icon for collapsing the navigation, as the current hamburger icon doesn't fit. Possible alternatives include a collapse icon or sidebar icon.

### Designs
[Figma artboards](https://www.figma.com/file/FStledDq3ytXssh4PykKf5/Combined-megamenu?type=design&node-id=648-15218&mode=design&t=o2VbXdFceTbxkrhU-0)
[Figma prototype](https://www.figma.com/proto/FStledDq3ytXssh4PykKf5/Combined-megamenu?page-id=648%3A15218&type=design&node-id=648-15642&t=o2VbXdFceTbxkrhU-0&scaling=min-zoom&starting-point-node-id=648%3A15601)

```[tasklist]
### Tasks
- [ ] https://github.com/grafana/grafana/issues/74916
- [ ] https://github.com/grafana/grafana/issues/74917
- [ ] https://github.com/grafana/grafana/issues/75273
- [ ] https://github.com/grafana/grafana/issues/75446
- [ ] https://github.com/grafana/grafana/issues/75725
- [ ] https://github.com/grafana/grafana/issues/75726
- [ ] https://github.com/grafana/grafana/issues/75727
- [ ] https://github.com/grafana/grafana/issues/76518
- [ ] https://github.com/grafana/grafana/issues/76519
- [ ] https://github.com/grafana/grafana/issues/77379
- [ ] https://github.com/grafana/grafana/issues/77263
- [ ] https://github.com/grafana/grafana/issues/77262
- [ ] https://github.com/grafana/grafana/issues/76520
- [ ] https://github.com/grafana/grafana/issues/81775
```

### LINKED ISSUE #81775: Remove `dockedMegaMenu` toggle and related code
now we've rolled out `dockedMegaMenu` in cloud, we can remove the toggle and default the behaviour to on for OSS
aim for 10.4.x

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- removes the `dockedMegaMenu` feature toggle
- removes old/unused code
  - removes the old `MegaMenu` folder
  - removes `SectionNav` now it's no longer used
  - tidies up the applinks in the backend 
- renames `DockedMegaMenu` folder to `MegaMenu`

**Why do we need this feature?**

- clean up 🧹 

**Who is this feature for?**

- everyone!

**Which issue(s) does this PR fix?**:

Fixes #81775 
Fixes #72446 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/locales/en-US/grafana.json [hunk 1]
```diff
@@ -880,9 +880,6 @@
       "subtitle": "Manage preferences across an organization",
       "title": "Default preferences"
     },
-    "performance-testing": {
```
Comment (JoaoSilvaGrafana) [inline public/locales/en-US/grafana.json:883]: I don't know if this is done differently now, but these translations still exist on other languages (german, spanish, etc.). Do we need to remove them here?

**Review by JoaoSilvaGrafana — COMMENTED** [review]
This is so pleasing 😌 
One question about translations, don't see anything else off to me 👏

File: public/locales/en-US/grafana.json (hunk 1 above)
Comment (ashharrison90) [reply public/locales/en-US/grafana.json:883]: great question - i don't _think_ so, i think what happens is:
- the PR gets merged
- the new english file gets uploaded to crowdin via the new action
- the translations are updated (in this case to remove that string)
- the new translation files are raised as a PR
- we merge the PR, cycle complete

cc @joshhunt @L-M-K-B to help confirm that process 🙇

Comment (joshhunt) [reply public/locales/en-US/grafana.json:883]: ahh. interesting. @L-M-K-B I guess this is a side effect of i18n-extract no longer writing to the other language files.

I think the flow you pointed out seems correct. Lets see how it goes 👍

**Review by JoaoSilvaGrafana — APPROVED** [review]
Lgtm then! 👏

**Pull request merged**